src/Table/Table.tsx

```diff
@@ -384,53 +387,60 @@ export class Table extends React.Component<Props, State> {
-                  rowClassNameFn && rowClassNameFn(rowData),
-                )}
-                key={rowIDFn(rowData)}
-                onClick={e => onRowClick && onRowClick(e, rowIDFn(rowData), rowData)}
-              >
-                {columns.map(({ props: col }: { props: any }) => (
-                  <Cell className={getCellClassName(col, rowData)} key={col.id} noWrap={col.noWrap}>
-                    {col.cell.renderer(rowData)}
-                  </Cell>
-                ))}
+      <>
+        <table className={classnames(cssClass.TABLE, fixed && cssClass.FIXED, className)}>
+          <Header
+            disableSort={disableSort}
+            onSortChange={columnID => this._toggleSort(columnID)}
+            sortState={sortState}
+          >
+            {columns}
+          </Header>
+          <tbody className={cssClass.BODY}>
+            {displayedData.length === 0 && !noDataContent ? (
+              <tr className={cssClass.ROW}>
+                <Cell className={cssClass.NO_DATA} colSpan={columns.length} noWrap>
+                  {!pageLoading && "NO DATA"}
+                </Cell>
               </tr>
-            ))
+            ) : (
+              displayedData.map(rowData => (
+                <tr
+                  className={classnames(
+                    cssClass.ROW,
+                    onRowClick && cssClass.CLICKABLE_ROW,
+                    rowClassNameFn && rowClassNameFn(rowData),
+                  )}
+                  key={rowIDFn(rowData)}
+                  onClick={e => onRowClick && onRowClick(e, rowIDFn(rowData), rowData)}
+                >
+                  {columns.map(({ props: col }: { props: any }) => (
+                    <Cell
+                      className={getCellClassName(col, rowData)}
+                      key={col.id}
+                      noWrap={col.noWrap}
+                    >
+                      {col.cell.renderer(rowData)}
+                    </Cell>
+                  ))}
+                </tr>
+              ))
+            )}
+          </tbody>
+          {paginated && (
+            <Footer
+              currentPage={displayedPage}
+              onPageChange={newPage => this.setCurrentPage(newPage)}
+              numColumns={columns.length}
+              numPages={numPages}
+              showLastPage={!lazy}
+              isLoading={pageLoading}
+              lengthUnknown={lazy && numRows == null && !allLoaded}
+            />
           )}
-        </tbody>
-        {paginated && (
-          <Footer
-            currentPage={displayedPage}
-            onPageChange={newPage => this.setCurrentPage(newPage)}
-            numColumns={columns.length}
-            numPages={numPages}
-            showLastPage={!lazy}
-            isLoading={pageLoading}
-            lengthUnknown={lazy && numRows == null && !allLoaded}
-          />
-        )}
-      </table>
+        </table>
+        {displayedData.length === 0 && noDataContent}
```

Comment:
Would you want this to be inside the `table`?

Reply:
That was my first idea. But I got a warning that we should only render `tr` elements inside a table. I can try passing `noDataContent` inside a cell in a row spanning all columns. Thanks for the suggestion